multi_physics/QED/kokkos_example/kokkos_example_commons.hpp

```diff
@@ -0,0 +1,351 @@
+            temp += std::isinf(field(i));},
+            num);
+    }
+    else if(check_nan && check_inf){
+        Kokkos::parallel_reduce("HowManyNaNsInfs_"+get_type_name<Real>(),
+        field.size(), KOKKOS_LAMBDA (const int& i, int& temp ) {
+            temp += std::isnan(field(i)) + std::isinf(field(i));},
+            num);
+    }
+
+    return (num == 0);
+}
+
+/**
+* Checks a 2D view with N,ParticleData<Real>::num_components
+* elements for NaN and/or infs
+*
+* @tparam Real the floating point type to be used
+* @param[in] field the view to be tested
+* @param[in] check_nan whether to check for NaNs in field
+* @param[in] check_inf whether to check for infinities in field
+* @return true if checks pass, false otherwise
+*/
+template <typename Real>
+bool check_multi(Kokkos::View<Real * [ParticleData<Real>::num_components]> vec,
+    const bool check_nan = true, const bool check_inf = false)
+{
+
+    int num = 0;
+
+    if(check_nan && !check_inf){
+        Kokkos::parallel_reduce("HowManyNaNsMulti_"+get_type_name<Real>(),
+        vec.extent(0), KOKKOS_LAMBDA (const int& i, int& temp ) {
+            for (int cc = 0; cc < ParticleData<Real>::num_components; ++cc){
+                temp += std::isnan(vec(i,cc));};},
+            num);
+    }
+    else if(!check_nan && check_inf){
+        Kokkos::parallel_reduce("HowManyInfsMulti_"+get_type_name<Real>(),
+        vec.extent(0), KOKKOS_LAMBDA (const int& i, int& temp ) {
+            for (int cc = 0; cc < ParticleData<Real>::num_components; ++cc){
+                temp += std::isinf(vec(i,cc));};},
+            num);
+    }
+    else if(check_nan && check_inf){
+        Kokkos::parallel_reduce("HowManyNaNsInfsMulti_"+get_type_name<Real>(),
+        vec.extent(0), KOKKOS_LAMBDA (const int& i, int& temp ) {
+            for (int cc = 0; cc < ParticleData<Real>::num_components; ++cc){
+                temp += std::isnan(vec(i,cc)) + std::isinf(vec(i,cc));};},
+            num);
+    }
+
+    return (num == 0);
+}
+
+using GenType = Kokkos::Random_XorShift64_Pool<>::generator_type;
+
+template <typename Real>
+struct get_rand{
+
+    /**
+    * Calls the drand() method of the Kokkos random number generator.
+    * Prevents results exaclty equal to 1.0
+    *
+    * @tparam Real the floating point type to be used
+    * @param[in,out] gen a random number generator
+    */
+    KOKKOS_INLINE_FUNCTION
+    static Real get(GenType& gen)
+    {
+        Real res = Real(1.0);
+        while(res >= Real(1.0))
+            res = gen.drand();
+        return res;
+    }
+};
+
+/**
+* Calls the frand() method of the Kokkos random number generator
+* (i.e. the single precision version). Prevents results exaclty equal to 1.
+*
+*/
+template<>
+struct get_rand<float>{
+    /**
+    * Calls the frand() method of the Kokkos random number generator
+    * (i.e. the single precision version). Prevents results exaclty equal to 1.
+    *
+    * @tparam Real the floating point type to be used
```

Comment:
There's not template parameter `Real` in this specialization.

Reply:
you are right!

Reply:
I have simplified and fused together this version and the double precision version of the function, making use of the new C++ 17 `if constexpr` feature.